src/resolver.js

```diff
@@ -46,15 +46,14 @@ exports.resolve = (binaryBlob, path, callback) => {
         // for the resolver
         node.links.forEach((l, i) => {
           const link = l.toJSON()
-          values[i] = {
+          // TODO by enabling something to resolve through link name, we are
+          // applying a transformation (a view) to the data, confirm if this
+          // is exactly what we want
+          values[i] = values[link.name] = {
```

Comment:
I don't think `values[link.name]` is needed here. The case is handled below.

Please also remove the "TODO".

Reply:
This was already in the code base although I don't think it was working correctly:
https://github.com/ipld/js-ipld-dag-pb/blob/fda23c444b5437fe15c2875dfe547ad1e63a2c27/src/resolver.js#L54-L57

It allows you to reference links by name via Links e.g. `/ipfs/QmHash/Links/myNamedLink`

Reply:
Got it. So this is kind of the IPFS vs. IPLD like traversal.

Reply:
Yeah, I don't know if it's _really_ necessary to be able to `/ipfs/QmHash/Links/myNamedLink` **as well as** `/ipfs/QmHash/myNamedLink` but the former was already there and we're adding the latter.

Reply:
`/ipfs/QmHash/Links/myNamedLink` is what I call IPLD-like traversal. It will then be use if you use the `/ipld` prefix.